gtsfm/bundle/bundle_adjustment.py

```diff
@@ -28,6 +41,78 @@ class BundleAdjustmentOptimizer:
     This class refines global pose estimates and intrinsics of cameras, and also refines 3D point cloud structure given
     tracks from triangulation."""
 
+    def __init__(self, shared_calib: bool = False):
+        """Initializes the parameters for bundle adjustment module.
+
+        Args:
+            shared_calib (optional): Flag to enable shared calibration across
+                                     all cameras. Defaults to False.
+        """
+
+        self._shared_calib = shared_calib
+
+    def __add_camera_prior_and_initial_value(
```

Comment:
maybe should be two separate functions?

`def __add_pose_prior()`
`def __initialize()`